src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp

```diff
@@ -181,6 +182,35 @@ bool ValuesBlockInputFormat::tryReadValue(IColumn & column, size_t column_idx)
     }
 }
 
+namespace
+{
+    void tryToReplaceNullFieldsInTupleWithDefaultValues(Field & value, const IDataType & type)
+    {
+        const DataTypeTuple * type_tuple = typeid_cast<const DataTypeTuple *>(&type);
+
+        if (!type_tuple || value.getType() != Field::Types::Tuple)
+            return;
+
+        Tuple & tuple_value = value.get<Tuple>();
+
+        size_t src_tuple_size = tuple_value.size();
+        size_t dst_tuple_size = type_tuple->getElements().size();
+
+        if (src_tuple_size != dst_tuple_size)
+            throw Exception("Bad size of tuple. Expected size: " + std::to_string(src_tuple_size) + ", actual size: " + std::to_string(dst_tuple_size), ErrorCodes::TYPE_MISMATCH);
```

Comment:
Ok. BTW, we also have format-like constructor for Exception.

Reply:
Fixed.